source/io/module_eggShell.js

```diff
@@ -460,6 +509,29 @@
             Data_WriteDouble(destination, value);
         };
 
+        Module.eggShell.dataWriteTypedArray = function (destination, value) {
+            var int32Byte = 4;
+            var rank = 1;
+            var newLengths = Module._malloc(rank * int32Byte);
```

Comment:
We need to deallocate this memory. Otherwise, it will become a memory leak.

Reply:
Consider the function may throw and the deallocation needs to happen before that.

Reply:
Fixed.